spec/CRM/Example/BooleanStateMachine.hs

```diff
@@ -0,0 +1,25 @@
+{-# LANGUAGE GADTs #-}
+
+module CRM.Example.BooleanStateMachine where
+
+import "crm" CRM.BaseMachine
+import "crm" CRM.StateMachine
+import "singletons-base" Data.Singletons.Base.TH
+
+booleanStateMachine
+  :: forall a
+   . SBool a
+  -> StateMachine Int Int
+booleanStateMachine initialState =
+  unrestrictedMachine @Bool
+    ( \state input -> case state of
+        SFalse ->
+          if even input
+            then ActionResult SFalse (input + 1)
+            else ActionResult STrue (input * 3)
+        STrue ->
+          if even input
+            then ActionResult STrue (input - 1)
+            else ActionResult SFalse (input * 5)
+    )
```

Comment:
I'm not sure I understand these state transitions.

Maybe this could be simplified to something meaningful, e.g. the Collatz function that remembers if the last number was even:

```suggestion
    ( \state input ->
    	ActionResult
    	  (bool (even input) SFalse STrue)
    	  (if even input
    	    then input `div` 2
    	    else input * 3 + 1)
```

That would make for a fun property test as well. :slightly_smiling_face:

Reply:
I completely made up some values for creating this machine.

I'll create an issue for coming up with better examples